### PR TITLE
throw on unsupported srs

### PR DESCRIPTION
**Pull request by mapsam**
This adds a check using gdal's `autoIdentifyEPSG` method. If you pass in an unsupported TIF srs, it returns an error. This throws before running gdal processes on unsupported srs's, which leads to a segfault - seen in https://github.com/mapbox/preprocessorcerer/issues/48

I'm not 💯 confident with this solution since we don't have a ton of fixtures of different SRS's. I'd be  hesitant to merge this without a 👍 on the confidence of the node-gdal API usage here.

**An option** to avoid `src.srs.autoIdentifyEPSG();` from acting on the object in use is to clone the entire object (separate but equal) and try there, so we don't accidentally change anything in the reprojected tiff that is returned.

cc @isiyu @rclark @GretaCB

### CODE REVIEW

**Comment by mapsam** [comment]
Ah, you know what - I'm thinking this might be better located in preprocessorcerer's `criteria` function instead. https://github.com/mapbox/preprocessorcerer/blob/master/preprocessors/tif-reproject.preprocessor.js#L24-L28

That's essentially doing similar checks, plus it doesn't affect the underlying data used in reprojection.

File: index.js [hunk 1]
```diff
@@ -37,6 +37,12 @@ function reproject(srcpath, dstpath) {
   var bandCount = src.bands.count();
   var dataType = src.bands.get(1).dataType;
 
+  try {
```
Comment (rclark) [inline index.js]: If you're just going to catch the error and throw it, there's no difference from just letting it happen. That is, I don't think you need to try/catch here to have the same effect.

**Comment by rclark** [comment]
I think it makes sense to do this here -- as you found failure can lead to a segfault / process crash. wmtiff shouldn't be that mean to any downstream callers.

I don't have any more context or experience to provide on the node-gdal usage...

File: index.js (hunk 1 above)
Comment (mapsam) [inline index.js]: You're right - removed!

**Comment by millzpaugh** [comment]
We're handling unsupported `srs` issues here: https://github.com/mapbox/preprocessorcerer/pull/92/files 

Closing this!

**Pull request closed without merging**